Controllers/ImageController.cs

```diff
@@ -0,0 +1,336 @@
+        //        {
+        //            var zipId = vals.FirstOrDefault();
+        //            if (zipId != null)
+        //            {
+        //                return Get(zipId);
+        //            }
+        //        }
+        //    }
+
+        //    return new StatusCodeResult((int)postResponse.StatusCode);
+        //}
+
+        
+        [HttpPost("generateBase64Images")]
+        public async Task<ActionResult> Base64([FromForm] ImageFormData Form)
+        {
+
+            using (var args = ImageGenerationModel.FromFormData(HttpContext.Request.Form, HttpContext.Request.Form.Files))
+            {
+                if (!string.IsNullOrEmpty(args.ErrorMessage))
+                {
+                    return new ObjectResult(args.ErrorMessage) { StatusCode = (int?)HttpStatusCode.BadRequest };
+                }
+
+                var imgs = GetProfilesFromPlatforms(args.Platforms)
+                    .Select(profile => new WebManifestIcon
+                    {
+                        Purpose = "any",
+                        Sizes = $"{profile.Width}x{profile.Height}",
+                        Src = CreateBase64Image(args, profile),
+                        Type = string.IsNullOrEmpty(profile.Format) ? "image/png" : profile.Format
+                    });
+
+                var response = new ObjectResult(JsonConvert.SerializeObject(imgs));
+                return response;
+            }
+        }
+
+        private static string ReadStringFromConfigFile(string filePath)
+        {
+            using (var fs = new FileStream(filePath, FileMode.Open, FileAccess.Read))
+            using (var sr = new StreamReader(fs))
+            {
+                return sr.ReadToEnd();
+            }
+        }
+
+        private IEnumerable<string> GetConfig(string platformId)
+        {
+            List<string> config = new List<string>();
+            string webRootPath = _webHostEnvironment.WebRootPath ?? _webHostEnvironment.ContentRootPath;
+            var root = Path.Combine(webRootPath, "App_Data");
+
+            string filePath = Path.Combine(root, platformId + "Images.json");
+            config.Add(ReadStringFromConfigFile(filePath));
+            return config;
+        }
+
+        private IReadOnlyList<Profile> GetProfilesFromPlatforms(IEnumerable<string> platforms)
+        {
+            List<Profile> profiles = null;
+            foreach (var platform in platforms)
+            {
+                // Get the platform and profiles
+                var config = GetConfig(platform);
+                if (config.Count() < 1)
+                {
+                    throw new HttpRequestException(HttpStatusCode.BadRequest.ToString());
+                }
+
+                foreach (var cfg in config)
+                {
+                    if (profiles == null)
+                    {
+                        profiles = JsonConvert.DeserializeObject<List<Profile>>(cfg);
```

Comment:
Is this config provided by the system? Do you need to guard against failure in reading here?

Reply:
Yep this is from system files we keeping in repo